staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -303,11 +305,23 @@ func (f *featureGate) GetAll() map[Feature]FeatureSpec {
 
 // Enabled returns true if the key is enabled.  If the key is not known, this call will panic.
 func (f *featureGate) Enabled(key Feature) bool {
```

Comment:
.Enabled is used in some really hot code paths... not sure we want all of this inside that method. When a feature gets added (via Add) or its state gets modified (via Set or SetFromMap), we can update the metrics then (a single time)

Reply:
I don't think every instance of this type should mutate global metrics by default. Just like a given feature gate set has to be surfaced via flags with a call to `myGateInstance.AddFlag(...)`, I think the containing binary should decide which feature gate set (if any) to hook up to global metrics.

Reply:
I agree with your assessment of the read path thing, but I disagree about making these metrics should be optional. We actually basically mandate feature enablement metrics in PRR and this basically just systemizes the process so that people don't have to do this in a one off way.